### ISSUE TITLE
Migration 'DbInitializer' does not finish with SQL Server

### ISSUE THREAD

**Issue by edikaufmann** [aspnet/Docs#8070]
thanks for your time.
when running the ContosoUniversity sample all runs fine up (and including) to 'Migrations!
In parallel, I started a new App (new Namespace) and it runs ok EXCEPT for the 'migrations' i.e. when:
Drop-Migration - add-migration - update-migration
- ONLY the Student table gets created!? 
-PM> never finishes -> see below log
- I do not see any errors

Hence: I have to update the DBs in SSMS 17.8. I use SQL server 17
- After the SSMS updates ALL works fine again
- I use for the PKs seeding (1 1)

PS: another small issue:: 'Download the completed app'. I never see in githup the 'tab' doenload the ^zip'. What do I do wrong!?!'

before I dig deeper perhaps somebody has seen this before!?
thanks for your help and ideas, ed

last statements in PM>

SELECT COUNT(*)
FROM [Student] AS [s]
Microsoft.EntityFrameworkCore.Database.Command:Information: Executed DbCommand (1ms) [Parameters=[@__p_0='?' (DbType = Int32), @__p_1='?' (DbType = Int32)], CommandType='Text', CommandTimeout='30']
SELECT [s].[StudentID], [s].[Cotisation], [s].[EnrollmentDate], [s].[FirstName], [s].[Health], [s].[LastName], [s].[Phone], [s].[eMail]
FROM [Student] AS [s]
ORDER BY [s].[LastName]
OFFSET @__p_0 ROWS FETCH NEXT @__p_1 ROWS ONLY
Microsoft.AspNetCore.Mvc.RazorPages.Internal.PageActionInvoker:Information: Executed action /Students/Index in 30.8331ms
Microsoft.AspNetCore.Hosting.Internal.WebHost:Information: Request finished in 42.6927ms 200 text/html; charset=utf-8
The thread 0x229c has exited with code 0 (0x0).
The thread 0x237c has exited with code 0 (0x0).
The thread 0x3af4 has exited with code 0 (0x0).
The thread 0x2068 has exited with code 0 (0x0).
The thread 0x3464 has exited with code 0 (0x0).
The thread 0x2d38 has exited with code 0 (0x0).
The thread 0x38e0 has exited with code 0 (0x0).
The thread 0x46e4 has exited with code 0 (0x0).

-----> does not finish with: PM>[Enter feedback here]

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 145343de-9267-9171-b6c9-718703b7544d
* Version Independent ID: 2c2f791f-d41c-f995-a941-2461151a7db4
* Content: [Razor Pages with EF Core in ASP.NET Core - Migrations - 4 of 8](https://docs.microsoft.com/en-gb/aspnet/core/data/ef-rp/migrations?view=aspnetcore-2.1&tabs=visual-studio)
* Content Source: [aspnetcore/data/ef-rp/migrations.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/data/ef-rp/migrations.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
Rather than using a different namespace you might have more luck if you create the same project in a different folder. Make sure ` "DefaultConnection": "Server=(localdb)\\mssqllocaldb;Database=` is different.

**Comment by edikaufmann** [comment]
the namespace is not the issue. My connection string points to a SQL server DB (this for several reasons).
My question is more in the direction of the PM> support when using SQL server. Remind you: all works just fine except the PM> 'update-database'. 
But well as I have to backup my DBs anyway I do some tests just to test the migration WITHOUT the 'PM> update-database i.e. just 'migrate' the models!

well I might come back if I find some more facts

**Comment by Rick-Anderson** [comment]
@ajcvickers do we have any guidance for running `update-database`  against a remote SQL server? If so I'd like to add that to the docs. 

CC @tdykstra

**Comment by Rick-Anderson** [comment]
@edikaufmann  did you see [Applying migrations in production](https://docs.microsoft.com/en-us/aspnet/core/data/ef-rp/migrations?view=aspnetcore-2.1&tabs=visual-studio#applying-migrations-in-production)

**Comment by ajcvickers** [comment]
@Rick-Anderson I'm not aware of any issues with using update-database against a remote server. 
@edikaufmann I would suggest filing an issue at https://github.com/aspnet/EntityFrameworkCore/issues with a runnable project/solution that demonstrates the behavior you are seeing.

**Comment by Rick-Anderson** [comment]
@edikaufmann  if you do file an issue of the EF repro, make sure you have the simplest possible repo that demonstrates the problem.

Obviously the remote SQL server must be configured correctly and you must have permission to update the DB.

**Comment by edikaufmann** [comment]
@Rick-Anderson
 I'm obviously having issues because I'm using SQL Server/SSMS. I still think it has to do with the 'migration business' (yes: I've read the 'Migration' chapter 4 in your Tutorial). BUT I need to learn some more on migration/SQL server before I can intelligently comment!?)
I.e I don't want to bother you guys before I know some more - that, in fact, was the reason I closed 8070 on August 14!

The interesting thing : the DB constraints/indexes are different from MSSQL to SQL server!? (this is (yet) a can of worms to me and probably my fault!).

Status on your Tutorial: I used your directions (NO download) and all works just fine, Thanks for your guys work!

As said on Aug 14 I will come back if I have some decent questions/comments to make on this subject!
regards and thanks again, ed
PS:  I leave it up to you to close 8070_
